docs/design/expressions/type_operators.md

```diff
@@ -0,0 +1,59 @@
+# Type operators
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Overview](#overview)
+-   [Details](#details)
+    -   [Precedence](#precedence)
+-   [Alternatives considered](#alternatives-considered)
+-   [References](#references)
+
+<!-- tocstop -->
+
+## Overview
+
+Carbon provides the following operators to transform types:
+
+-   `const` as a prefix unary operator produces a `const`-qualified type.
+-   `*` as a postfix unary operator produces a pointer _type_ to some other
+    type.
+
+The pointer type operator is also covered as one of the
+[pointer operators](pointer_operators.md).
+
+## Details
+
+The semantic details of both `const`-qualified types and pointer types are
+provided as part of the [values](/docs/design/values.md) design:
+
+-   [`const`-qualified types](/docs/design/values.md#const-qualified-types)
+-   [Pointers](/docs/design/values.md#pointers)
+
+The syntax of these operators tries to mimic the most common appearance of
+`const` types and pointer types in C++.
+
+### Precedence
+
+Because these are type operators, they don't have any precedence relationship
+with non-type operators. `const` binds more tightly than `*` and can appear
```

Comment:
Following up on my comment in expressions/README.md, we need a precedence relationship between type operators and `as`, and I think that type operators should bind more tightly than `as`.

Reply:
Done and documented both `as` and the desired transitive use case of `if`.